source/core/NumericString.cpp

```diff
@@ -2090,7 +2093,10 @@ struct ByteArrayToStringParamBlock : InstructionCore
 static bool CheckUnsupportedEncodingError(ErrorCluster* errorCluster, UInt16 stringEncoding, TypedArrayCoreRef outputArray) {
     if (stringEncoding != 0) {
         if (errorCluster != nullptr) {
-            errorCluster->SetErrorAndAppendCallChain(true, 1, "Encoding type not supported");
+            errorCluster->SetErrorAndAppendCallChain(true, kCannotConvertErr, "String To Byte Array");
```

Comment:
CheckUnsupportedEncodingError appears to be called in both ByteArrayToString and StringToByteArray, but here we are just reporting the error as always happening with "String To Byte Array". Consider adding a parameter and pass the name to avoid this problem.

Reply:
Done in update 3.